src/engraving/dom/measure.cpp

```diff
@@ -533,8 +533,11 @@ bool Measure::showsMeasureNumberInAutoMode()
         return false;
     }
 
+    int interval = style().styleI(Sid::measureNumberInterval);
+
     // Measure numbers should not show on first measure unless specified with Sid::showMeasureNumberOne
-    if (!no()) {
+    // except, when showing numbers on each measure, and first measure is after anacrusis - then show always
+    if (isFirst() || (prevMeasure() && prevMeasure()->irregular() && prevMeasure()->isFirst() && interval != 1)) {
```

Comment:
This doesn't look optimal. We may call prevMeasure() up to 5 times here... You should call it only once